packages/sqlite_async/lib/src/native/database/connection_pool.dart

```diff
@@ -88,6 +90,11 @@ class SqliteConnectionPool with SqliteQueries implements SqliteConnection {
       return;
     }
 
+    if (_availableReadConnections.isEmpty && _runningWithAllConnections) {
+      // Wait until withAllConnections is done
```

Comment:
Can you expand this comment to explain why? If I remember correctly, we don't want to spawn a new connection while `withAllConnections` is active because that could mean the new connection does not end up in the callback to readers? That's something mentioning here I think.